### PR TITLE
fix(security): Add semantic-release to dev dependencies

### PR DESCRIPTION
**Pull request by AaronDewes**
Installing it during the actions run makes supply chain attacks by a possibly compromised semantic-release package easier. Pinning the version through the lockfile prevents this.

In addition, this prevents breaking changes in semantic-release breaking the build.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -20,10 +20,8 @@ jobs:
         with:
           node-version: lts/*
           cache: npm
-      - run: >-
-          npm install -D semantic-release
-          semantic-release-plugin-update-version-in-files
-      - run: npx semantic-release --debug
+      - run: npm install --frozen-lockfile
```
Comment (G-Rath) [inline .github/workflows/release.yml:23]: This is a yarn flag - you want `npm ci`

```suggestion
      - run: npm ci
```

**Review by G-Rath — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (AaronDewes) [reply .github/workflows/release.yml]: Thanks!

**Review by gr2m — COMMENTED** [review]
you can pin the version with when calling with npx instead, I think @travi added a renovate strategy to update 

We are also very careful and put a lot of safety mechanisms into @semantic-release to prevent supply chain attacks. 

And lastly, we don't have any CI checks for semantic release. In the end we will get updates for semantic release and merge them in anyway without much review. I don't think it's worth it.

**Comment by wolfy1339** [comment]
Currently, semantic-release is using older versions of Octokit, which conflicts with the newer versions of Octokit which require the newer versions that are ESM.

Installing it on the fly with npx and removing it from the dev dependencies unblocked release workflows

**Pull request closed without merging**